sycl/test/on-device/online_compiler/online_compiler_common.hpp

```diff
@@ -0,0 +1,181 @@
+     CGH.set_arg(1, OutputBuf.get_access<sycl::access::mode::write>(CGH));
+     CGH.parallel_for(sycl::range<1>{N}, Kernel);
+   }).wait();
+
+  auto Out = OutputBuf.get_access<sycl::access::mode::read>();
+  for (int I = 0; I < N; I++)
+    std::cout << I << "*2 + 100 = " << Out[I] << "\n";
+}
+#endif // RUN_KERNELS
+
+int main(int argc, char **argv) {
+  cl::sycl::queue Q;
+  cl::sycl::context Context = Q.get_context();
+  cl::sycl::device Device = Q.get_device();
+
+  { // Compile and run a trivial OpenCL kernel.
+    std::cout << "Test case1\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(
+          std::string(CLSource),
+          // Pass two options to check that more than one is accepted.
+          std::vector<std::string>{std::string("-cl-fast-relaxed-math"),
+                                   std::string("-cl-finite-math-only")});
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+#ifdef RUN_KERNELS
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+#endif // RUN_KERNELS
+  }
+
+  { // Compile and run a trivial OpenCL kernel using online_compiler()
+    // constructor accepting SYCL device.
+    std::cout << "Test case2\n";
+    online_compiler<source_language::opencl_c> Compiler(Device);
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(std::string(CLSource));
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+#ifdef RUN_KERNELS
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+#endif // RUN_KERNELS
+  }
+
+#ifdef COMPILE_CM_KERNEL
+  // TODO: this test is temporarily turned off because CI buildbots do not set
+  // PATHs to clangFEWrapper library properly.
+  { // Compile a trivial CM kernel.
+    std::cout << "Test case3\n";
+    online_compiler<source_language::cm> Compiler;
+    try {
+      std::vector<byte> IL = Compiler.compile(std::string(CMSource));
+
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+  }
+#endif // COMPILE_CM_KERNEL
+
+  { // Compile a source with syntax errors.
+    std::cout << "Test case4\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    bool TestPassed = false;
+    try {
+      IL = Compiler.compile(std::string(CLSourceSyntaxError));
```

Comment:
```suggestion
      IL = Compiler.compile(CLSourceSyntaxError);
```

Reply:
Fixed in 893852b81738ba82be3748c1cfcbabc8f61bd94d